# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/DocAccessible.h
# Commit: 3cfe5bd5d084
# Full Hash: 3cfe5bd5d084baf18f5d95b0e879f6fc179fac34
# Author: Morgan Reschenberg <mreschenberg@mozilla.com>
# Date: 2021-10-06 04:09:04
# Regressor Bug: 1726227
# File Overlap Count: 1
# Description:
#   Bug 1726227: Cache parent-relative accessible bounds, resolution in parent process r=Jamie,eeejay
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123399
# ==============================================================================

diff -r 76a965804210 -r 3cfe5bd5d084 accessible/generic/DocAccessible.h
--- a/accessible/generic/DocAccessible.h	Tue Oct 05 20:24:03 2021 +0000
+++ b/accessible/generic/DocAccessible.h	Tue Oct 05 20:44:46 2021 +0000
@@ -17,6 +17,7 @@
 #include "nsIDocumentObserver.h"
 #include "nsIObserver.h"
 #include "nsITimer.h"
+#include "nsTHashSet.h"
 #include "nsWeakReference.h"
 
 class nsAccessiblePivot;
@@ -340,6 +341,13 @@
   void ContentRemoved(LocalAccessible* aAccessible);
   void ContentRemoved(nsIContent* aContentNode);
 
+  /*
+   * Add the given accessible to mMaybeBoundsChanged so we
+   * can (later) check if its bounds have changed, and update
+   * the cache appropriately.
+   */
+  void MarkForBoundsProcessing(LocalAccessible* aAcc);
+
   /**
    * Updates accessible tree when rendered text is changed.
    */
@@ -483,6 +491,13 @@
   void ProcessInvalidationList();
 
   /**
+   * Called from NotificationController to process this doc's
+   * mMaybeBoundsChanged list. Sends a cache update for each acc in this
+   * doc whose bounds have changed since reflow.
+   */
+  void ProcessBoundsChanged();
+
+  /**
    * Steals or puts back accessible subtrees.
    */
   void DoARIAOwnsRelocation(LocalAccessible* aOwner);
@@ -697,6 +712,8 @@
 
   // Exclusively owned by IPDL so don't manually delete it!
   DocAccessibleChild* mIPCDoc;
+
+  nsTHashSet<RefPtr<LocalAccessible>> mMaybeBoundsChanged;
 };
 
 inline DocAccessible* LocalAccessible::AsDoc() {